### DISCUSSION TITLE
Any ETA on Node Groups?

### DISCUSSION THREAD

**Discussion by dibli-goost** [bnpr/Malt#371] [Feedback & Feature Requests]
I've tried keeping up with #348, I don't think I've seen any discussion about Node Groups there.

Personally finding it very frustrating to not have them, as node trees get bigger and bigger it gets hard to manage without. I think it's been suggested here prior that we should code our own shader nodes instead at that point, but I don't think this will truly solve the issue.

A big problem I have with it, is that asset linking is a much better alternative in my case. I put my big shader Node Group in a project somewhere in my asset library, then link it to other projects where necessary. This means it's locatable in the asset browser with all other assets I use. I don't have to finagle with GLSL code and can instead just noodle around with the nodes, changes reflected in that asset project file are reflected in every linked file automatically.

Having my shader be a part of the add menu is not super important to me either, as I'd prefer that menu being as bare bones as possible. So adding custom nodes to Malt isn't really super appealing to me just yet.

Given that the screen pass node exists, it looks like the framework for node groups is kinda there. A lot of the improvements y'all have made in #348 have been towards better matching EEVEE/Cycles workflows. I think Node Groups are the next step there.

**Comment by Kolupsy** [comment]
I think node groups should be added at some point, sooner than later if possible. you CAN write your own shaders and they certainly have their advantages but I do agree that being able to organize and reuse node groups is a very powerful and already standard workflow. I would love to support it myself but unfortunately I dont have that much knowledge about how malt compiles node trees just yet.

**Comment by pragma37** [comment]
I know node groups are important and they are high on my priority list.
It's true that the system needed for it is somewhat implemented already but it would still require some work to add the feature.

Another issue is how to handle groups and node tree type compatibility, since the available nodes depend on the type of node tree.

> A big problem I have with it, is that asset linking is a much better alternative in my case. I put my big shader Node Group in a project somewhere in my asset library, then link it to other projects where necessary. This means it's locatable in the asset browser with all other assets I use. I don't have to finagle with GLSL code and can instead just noodle around with the nodes, changes reflected in that asset project file are reflected in every linked file automatically.

You can add your GLSL code to a global plugin and it will be available automatically on every .blend file. It even gets reloaded automatically while you are running Blender.

The actual advantages of node groups, IMO, are that you don't need to code to use them and that some parameter defaults can't be set directly from GLSL functions (Images and Color Ramps).
